druid-shell/src/alert.rs

```diff
@@ -0,0 +1,451 @@
+/// **Delete file** make it much more clear what will happen compared to a generic **OK**,
+/// **Confirm**, or **Yes**.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct AlertButton {
+    pub(crate) label: Cow<'static, str>,
+}
+
+impl AlertToken {
+    /// A token that does not correspond to any alert.
+    pub const INVALID: AlertToken = AlertToken(0);
+
+    /// Create a new token with the specified `id`.
+    pub(crate) const fn new(id: usize) -> AlertToken {
+        AlertToken(id)
+    }
+}
+
+impl AlertResponse {
+    /// Create a new alert response.
+    pub(crate) fn new(token: AlertToken, button: Option<AlertButton>) -> AlertResponse {
+        AlertResponse { token, button }
+    }
+
+    /// Returns the [`AlertToken`] that identifies the alert.
+    ///
+    /// [`AlertToken`]: struct.AlertToken.html
+    #[inline]
+    pub fn token(&self) -> AlertToken {
+        self.token
+    }
+
+    /// Returns the [`AlertButton`] that closed the alert.
+    ///
+    /// # Canceled
+    ///
+    /// This will return `None` if the alert was canceled.
+    ///
+    /// Keep in mind that the platform may forcefully cancel the alert, e.g. when
+    /// the parent window is closed, even when the alert dialog has no cancel button.
+    ///
+    /// [`AlertButton`]: struct.AlertButton.html
+    #[inline]
+    pub fn button(&self) -> Option<&AlertButton> {
+        self.button.as_ref()
+    }
+}
+
+impl AlertButton {
+    /// A primary button with the English label **OK**.
+    pub const OK: AlertButton = AlertButton::new("OK");
+    /// A cancel button with the English label **Cancel**.
+    pub const CANCEL: AlertButton = AlertButton::new("Cancel");
+
+    /// Create a new `const` button with the specified `label`.
+    ///
+    /// You should give a short but descriptive label like **Send message** that lets
+    /// the user know what will happen when they click the button.
+    ///
+    /// See [`AlertButton`] for more information.
+    ///
+    /// [`AlertButton`]: struct.AlertButton.html
+    pub const fn new(label: &'static str) -> AlertButton {
+        AlertButton {
+            label: Cow::Borrowed(label),
+        }
+    }
+
+    /// Create a new button with the specified `label` at runtime.
+    ///
+    /// This is the runtime version of [`new`] that allows you to use
+    /// runtime generated labels, e.g. dynamic personalization or translation.
+    ///
+    /// You should give a short but descriptive label like **Send message** that lets
+    /// the user know what will happen when they click the button.
+    ///
+    /// See [`AlertButton`] for more information.
+    ///
+    /// [`new`]: #method.new
+    /// [`AlertButton`]: struct.AlertButton.html
+    pub fn dynamic(label: impl Into<String>) -> AlertButton {
+        AlertButton {
+            label: Cow::Owned(label.into()),
+        }
+    }
+}
+
+impl Default for AlertOptions {
+    /// Create a default set of alert options.
```

Comment:
I think a short overview of what default means for `AlertOptions` would be good. I.e. no icon and only an OK button.